### ISSUE TITLE
Request Failed

### ISSUE THREAD

**Issue by benjaminwnelson** [MichelNivard/gptstudio#72]
When I try to use the package I get the following error. Do you need a paid version?

` OpenAI API request failed [429].Error message: You exceeded your current quota, please check your plan and billing details.
Warning: Error in purrr::map: ℹ In index: 3.
Caused by error in `if (.x$role == "system") ...`:
! argument is of length zero
  95: <Anonymous>
  94: signalCondition
  93: signal_abort
  92: rlang::abort
  91: cli::cli_abort
  90: h
  89: .handleSimpleError
  82: vctrs_vec_compat
  81: map_
  80: where_if
  79: discard
  78: purrr::compact
  76: make_chat_history
  75: ::
shiny
observe [/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/gptstudio/shiny/app.R#82]
  74: <observer>
   3: runApp
   2: print.shiny.appobj
   1: <Anonymous>`

**Comment by JamesHWade** [comment]
Hmmm... I'm not sure why the request failed. This does show me we need better error handling. Were you able to find a workaround?

**Comment by solvsa** [comment]
This seems to be a useful addin! However, I have encountered the same problem. Have not found a workaround yet.

**Comment by arpatoz** [comment]
Same here..

`! OpenAI API request failed [429].Error message: You exceeded your current quota, please check your plan and billing details.
$error
$error$message
[1] "You exceeded your current quota, please check your plan and billing details."

$error$type
[1] "insufficient_quota"

$error$param
NULL

$error$code
NULL`

**Comment by MichelNivard** [comment]
You all need to start paying OpenAI for the use of the API if you want to
continue using it, you can set up payment on their website.

**Comment by calderonsamuel** [comment]
Indeed. This error suggest that the owner of the api key reached the quota limit. There is an article in the openai docs explaining this error. https://help.openai.com/en/articles/6891831-error-code-429-you-exceeded-your-current-quota-please-check-your-plan-and-billing-details 

Maybe `gpt_edit()` or an internal can suggest the link to the documentation?

**Comment by JamesHWade** [comment]
Good idea on linking to documentation.

**Comment by ManuelSpinola** [comment]
I got the same error, however when I sign in in openaAI and use the API I can use it without problems.
